lips/lip-6.mdx

```diff
@@ -0,0 +1,253 @@
+```
+struct ValidatorInfo {
+  addr: address,
+  consensus_voting_power: u64,
+  config: ValidatorConfig::Config,
+  last_config_update_time: u64,
+}
+```
+
+## Transaction scripts
+The following transaction scripts can be run by the LibraRoot:
+* [`create_validator_account`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/create_validator_account.move): creates a Validator Owner account (an account with Validator role)
+* [`create_validator_operator_account`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/create_validator_operator_account.move) - to create a Validator Operator account
+* [`add_validator_and_reconfigure`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/add_validator_and_reconfigure.move) - to add Validator Owner to the Validator Set with its current config and trigger reconfiguration
+* [`remove_validator_and_reconfigure`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/remove_validator_and_reconfigure.move) - to remove Validator Owner from the Validator Set and trigger reconfiguration
+
+The following transaction scripts can be run by the Validator Owner:
+* [`set_validator_operator`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/set_validator_operator.move) - to designate the Validator Operator
+
+The following transaction scripts can be run by the Validator Operator:
+* [`set_validator_config_and_reconfigure`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/set_validator_config_and_reconfigure.move) - to change values of the Validator Config resource and update the config in the Validator Set (aborts if the Validator Owner is not in the Validator Set)
+
+The following script relevant to validator's operations can also be run by any LibraAccount's owner:
+* [`rotate_authentication_key`](https://github.com/libra/libra/blob/master/language/stdlib/transaction_scripts/rotate_authentication_key.move) - to rotate the authentication key of the Validator Owner Account
+
+# Workflows
+
+This section details the workflows or common scenarios related to Validator Set management.
+## Validator Owner and Validator Operator accounts creation
+LibraRoot (through the Association Operation Services, "AOS") creates Validator Owner and Validator Operator accounts. Currently, Validator Owners are not maintaining ownership of their Libra accounts. The LibraRoot account sends Validator Owners' transaction on their behalf via AdminScripts. The Validator Owner's account address is derived from the name of the Validator Owner, the auth-key is all-zeroes vector.n The LibraRoot creates the Validator Owner's account by sending the `create_validator_account` transaction script.
```

Comment:
Once this LIP moves to Approved stage, it cannot be changed. Therefore, I don't think it is a good idea to include comments on temporary situations, like the current owners lack of key custody.

Reply:
I don't have a strong opinion here, but I see the concern of @davidiw in that this LIP is currently the only place to describe how the system is run and if we omit it from here, then this major caveat is likely to go under the radars.

What is the procedure to update a LIP? In case e.g. the design of the validator set changes? Could these lips have versions (similar to RFC, NIST standards)?

One more question: do we plan to have a LIP on how the AOS is being run? I.e. who are the approvers, how HSMs are involved, etc.?

Reply:
I'll talk to David Dill tomorrow, Thu, to remove the TODO.

Reply:
I think this might be a bit too much detail though. I think we should speak on the hand of what the abstraction does here even if the implementation isn't quite there.

I don't want to block progress on the PR, so I'll approve this PR but also leave this blurb (that can be updated in a later PR or even by me!)

LibraRoot creates good...

What is required to create an account? A public key and a name (not there)

However Owners may desire LibraRoot to maintain the accounts on their behalf ...

How are account addresses derived ...

Reply:
@valerini I agree with @davidiw  's approach. we can say "owner may delegate to LibraRoot" or "LibraRoot may initially create owner account with auto delegation to LibraRoot and revoke it when Owners notify the association they have established custody of their own keys".

Reply:
Sounds good, I tried to rephrase this large paragraph, hope it looks better now.